logdna/resource_view.go

```diff
@@ -377,6 +416,23 @@ func resourceView() *schema.Resource {
 						"bodytemplate": {
 							Type:     schema.TypeString,
 							Optional: true,
+              // This function compares JSON, ignoring whitespace that can occur in a .tf config.
+              // Without this, `terraform apply` will think values are different from remote to state.
+              DiffSuppressFunc:func(k, old, new string, d *schema.ResourceData) bool {
+                var jsonOld, jsonNew interface{}
+                var err error
+                err = json.Unmarshal([]byte(old), &jsonOld)
+                if err != nil {
+                  return false
+                }
+                err = json.Unmarshal([]byte(new), &jsonNew)
+                if err != nil {
+                  return false
+                }
+                isDiff := reflect.DeepEqual(jsonNew, jsonOld)
+                log.Println("[DEBUG] BodyTemplate value in state appears the same as remote?", isDiff)
+                return isDiff
+              },
```

Comment:
On second thought, this is probably not needed if we use the value that's unmarshalled during the channel parsing and just assign it to the data at that point (or throw an error). 

This suppression function was fun to play around with, though.  It at least proved why I was seeing diffs between the config and when it does `terraform refresh`

Reply:
See the other comment.  Not gonna do this.